### PR TITLE
[[ Bug 19058 ]] Remove use of obj-c categories from the engine

### PR DESCRIPTION
**Pull request by runrevmark**

### CODE REVIEW

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 64385fe

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 64385fe075d3c3a21543aff89657f6e2c3903504

**Comment by livecode-vulcan** [comment]
😞 test failure 64385fe075d3c3a21543aff89657f6e2c3903504

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/779)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/784)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/777)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/778)
* try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/784)
* try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/776)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/776)

**Comment by runrevmark** [comment]
@livecodeali: As this is an internal change, I'm not sure it needs a release note?

**Comment by livecodeali** [comment]
@runrevmark Fair enough - I was thinking 'has bug number' -> 'needs release note', but actually everything should have a bug number.

**Comment by livecodeali** [comment]
@livecode-vulcan retry 64385fe

**Comment by livecode-vulcan** [comment]
😞 test failure 64385fe075d3c3a21543aff89657f6e2c3903504

* try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/801)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/806)
* try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/799)
* try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/800)
* try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/806)
* try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/798)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/798)

File: engine/src/mbliphonedialog.mm [hunk 1]
```diff
@@ -161,7 +161,7 @@ static void dopopupanswerdialog_prewait(void *p_context)
             for(uint32_t i = 0; i < ctxt -> button_count; i++)
             {
                 UIAlertAction *t_action;
-                t_action = [UIAlertAction actionWithTitle: [NSString stringWithMCStringRef: ctxt -> buttons[i] ]
+                t_action = [UIAlertAction actionWithTitle: MCStringConvertToAutoreleasedNSString(ctxt -> buttons[i) ]
                                                     style: UIAlertActionStyleDefault
```
Comment (livecodepanos) [inline engine/src/mbliphonedialog.mm]: This should be 

`t_action = [UIAlertAction actionWithTitle: MCStringConvertToAutoreleasedNSString(ctxt -> buttons[i]) `

**Comment by runrevmark** [comment]
@livecode-vulcan review not okay 64385fe

**Comment by livecode-vulcan** [comment]
💔 review by @runrevmark not ok 64385fe075d3c3a21543aff89657f6e2c3903504

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok 64385fe

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok 64385fe075d3c3a21543aff89657f6e2c3903504

**Comment by livecodepanos** [comment]
@livecode-vulcan retry 64385fe

**Comment by livecode-vulcan** [comment]
😞 test failure 64385fe075d3c3a21543aff89657f6e2c3903504

* try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/886)
* try-community-mac: [pending](https://vulcan.livecode.com/builders/try-community-mac/builds/892)
* try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/884)
* try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/885)
* try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/892)
* try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/883)
* try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/883)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok c47ce1f

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok c47ce1f8ddd16d62d99474e59214a81695c3eb69

**Comment by livecode-vulcan** [comment]
😞 test failure c47ce1f8ddd16d62d99474e59214a81695c3eb69

* try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/887)
* try-community-mac: [pending](https://vulcan.livecode.com/builders/try-community-mac/builds/893)
* try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/885)
* try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/886)
* try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/893)
* try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/884)
* try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/884)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok 6ccf67c

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok 6ccf67c29e31b0d5431d0ba8171d00c1768c3a3f

**Comment by livecode-vulcan** [comment]
😎 test success 6ccf67c29e31b0d5431d0ba8171d00c1768c3a3f

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/888)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/894)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/886)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/887)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/894)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/885)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/885)

**Pull request merged**